src/Domain/Orientation.hpp

```diff
@@ -0,0 +1,88 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Orientation.
+
+#pragma once
+
+#include <array>
+#include <iosfwd>
+#include <memory>
+
+#include "Domain/Direction.hpp"
+#include "Domain/SegmentId.hpp"
+
+/*!
+ * \ingroup DomainCreators
+ * \brief Allows one to construct a custom Orientation between two Blocks.
```

Comment:
It actually _is_ the custom orientation between blocks, but having the docs for Orientation be "a custom Orientation between two blocks" doesn't seem very useful.  How about something like "Describes relative axis alignment between two blocks"?